src/rust/engine/src/nodes.rs

```diff
@@ -1248,7 +1248,11 @@ impl Display for NodeKey {
       &NodeKey::Scandir(ref s) => write!(f, "Scandir({})", (s.0).0.display()),
       &NodeKey::Select(ref s) => write!(f, "{}", s.product),
       &NodeKey::Task(ref task) => {
-        write!(f, "@rule {}({})", task.task.display_info.name, task.params)
+        //TODO cf. https://github.com/pantsbuild/pants/issues/7907 , we probably
```

Comment:
Nit: The convention is `TODO(#7907)`. Would be good to fix so that people can find it via Grep.

Reply:
Also I think each comment is supposed to have a single space between the `//` and the message.